src/main/java/com/ezasm/parsing/Line.java

```diff
@@ -59,6 +61,10 @@ public Line(String instruction, String[] arguments) throws ParseException {
                 this.arguments[i] = new ImmediateInput(
                         Conversion.longToBytes(Lexer.getCharacterImmediate(arguments[i])));
             } else if (Lexer.isRegister(arguments[i])) {
+                // Get the Stored Register Number
+                if (registerNum == 0) {
```

Comment:
We cannot guarantee that the first register found will be an output for every instruction so this does not work for those odd instructions. A different approach could be to make changes to the `set` functions in the `Register` class which trigger the highlighting.